packages/core/src/dialects/momento/query-interface-typescript.ts

```diff
@@ -0,0 +1,406 @@
+      for (const cache of listCaches.getCaches()) {
+        caches.push({ tableName: cache.getName() });
+      }
+    }
+
+    return caches;
+  }
+
+  async dropAllTables(options?: QiDropAllTablesOptions) {
+    const tables = await this.showAllTables(options);
+    const conn = await this.sequelize.connectionManager.getConnection() as MomentoConnection;
+
+    for (const table of tables) {
+      // eslint-disable-next-line no-await-in-loop
+      const response = await conn.cacheClient.deleteCache(table.tableName);
+      if (response instanceof DeleteCache.Error && response.errorCode() !== MomentoErrorCode.NOT_FOUND_ERROR) {
+        throw new Error(`An exception occured while deleting cache: ${response.message()}`);
+      }
+    }
+  }
+
+  async dropTable(tableNameOrModel: TableName,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    options?: QiDropTableOptions) {
+    const tableNameObject = isModelStatic(tableNameOrModel) ? tableNameOrModel.getTableName()
+      : isString(tableNameOrModel) ? { tableName: tableNameOrModel }
+        : tableNameOrModel;
+    const conn = await this.sequelize.connectionManager.getConnection() as MomentoConnection;
+
+    const response = await conn.cacheClient.deleteCache(tableNameObject.tableName);
+
+    if (response instanceof DeleteCache.Error && response.errorCode() !== MomentoErrorCode.NOT_FOUND_ERROR) {
+      throw new Error(`An exception occured while deleting cache: ${response.message()}`);
+    }
+  }
+
+  // Ideally the data-types.ts file should have this logic for individual data types
+  // and Sequelize should be calling it before insert or select to handle serialization or deserialization.
+  // It seems they are not, and is hidden in the abstraction somewhere. This is probably acceptable for v0
+  // and we should explore piggybacking on Sequelize for the same.
+  serializeAttribute(key: string, attribute: NormalizedAttributeOptions, value: any) {
+    const type = attribute?.type;
+    let serializedValue;
+    if (type !== undefined) {
+      const typeString = type.toString();
+      switch (typeString) {
+        case 'VARCHAR(255)':
+          serializedValue = String(value);
+          break;
+        case 'DATETIME':
+        case 'DATETIME(6)':
+          if (value instanceof Date) {
+            serializedValue = value.toISOString();
+            break;
+          }
+
+          throw new Error('Datetime must be an instance of a date');
+        case 'INTEGER':
+          serializedValue = String(value);
+          break;
+        case 'FLOAT':
+          serializedValue = String(value);
+          break;
+        case 'BOOLEAN':
+          serializedValue = String(value);
+          break;
+        default:
+          throw new Error('Supported types are DATETIME, INTEGER, VARCHAR, BOOLEAN, AND FLOAT');
+      }
+    }
+
+    return serializedValue;
+  }
+
+  deserializeAttribute(key: string, attribute: NormalizedAttributeOptions, value: string) {
```

Comment:
this is interesting. I think good for now, but if we get into any conversations about schema changes, I think we might ultimately need to store the type information as part of the serialized payload, maybe using protobuf or something. to ensure that we can deserialize existing data even if the schema changes. but that's a problem for the future :)